wallet/src/error.rs

```diff
@@ -114,20 +109,31 @@ pub enum Error {
     StorageIsEncrypted,
     /// Tokio task join error
     #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
     TaskJoin(#[from] tokio::task::JoinError),
     /// Transaction not found
     #[error("transaction {0} not found")]
     TransactionNotFound(TransactionId),
 }
 
-/// Use this to serialize Error variants that implements Debug but not Serialize
-fn display_string<T, S>(value: &T, serializer: S) -> std::result::Result<S::Ok, S::Error>
-where
-    T: Display,
-    S: Serializer,
-{
-    value.to_string().serialize(serializer)
+// Serialize type with Display error
+impl Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_map(Some(2))?;
+        let kind_dbg = format!("{self:?}");
+        // Safe to unwrap because kind_dbg is never an empty string
+        let kind = kind_dbg
+            .split_whitespace()
+            .next()
+            .unwrap()
+            .split('(')
+            .collect::<Vec<&str>>()[0];
```

Comment:
You can probably `.next().unwrap()` again instead of collecting, less allocations